lib/browser-specific.js

```diff
@@ -194,42 +195,6 @@ function scoreSubtests(browserSubtests) {
     browserSubtests.forEach(s => s.moveNext());
   }
 
-  // At this point, at least one browser is out of subtests. The remaining
-  // subtests in other browsers do constitute valid tests, so we need to
-  // increment the denominator. If exactly one browser is out of subtests, there
-  // could also be remaining browser specific failures.
-  while (browserSubtests.some(subtest => subtest.hasCurrent())) {
-    const [name] = findSmallestNameAndIndex(browserSubtests);
-    const onSameSubtest = browserSubtests.filter(
-        s => s.hasCurrent() && s.value().name == name);
-    if (onSameSubtest.length >= browserSubtests.length) {
-      // This should not happen; the code above should only finish once some
-      // browser is out of subtests.
-      throw new Error('Internal error: Previous loop terminated too early');
-    }
-
-    if (name !== prevName) {
-      // If N-1 browser have this subtest with a pass status, then the one where
-      // it is missing is a browser-specific failure.
-      denominator += 1;
-      if (onSameSubtest.length == browserSubtests.length - 1) {
-        if (onSameSubtest.every(s => TEST_PASS_STATUSES.includes(
-            s.value().status))) {
-          for (let i = 0; i < browserSubtests.length; i++) {
-            if (!browserSubtests[i].hasCurrent()) {
-              counts[i] += 1;
-              break;
-            }
-          }
-        }
-      }
-    }
-
-    // Move all with the smallest name on.
-    prevName = name;
-    onSameSubtest.forEach(subtest => subtest.moveNext());
```

Comment:
Looking at this again, I do wonder if we need to change moving the iterator in the first commit. Hopefully the tests give that away?

Reply:
No, because we've continued in the case where we have fewer results.